### PR TITLE
feature/user-whitelisting

### PR DESCRIPTION
**Pull request by dcepil**
added `!whitelist`:
- allows users with a `Moderator` role to add/remove users on the whitelist by calling `!whitelist @User`
- an arg-less call will fetch the currently whitelisted users and send a list of them to the command caller, through the DMs
- to support this implementation, [`axios`](https://www.npmjs.com/package/axios) has been added as a dependency

updated `user.js`:
- added a `findOrCreateByDiscordId(id)` method, which fetches a user from the database with a matching `discordId` by the passed `id`

### CODE REVIEW

File: commands/whitelist.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+const setWhitelist = async (msg, args) => {
+  if (
+    msg.channel.type === "text" &&
+    msg.member.roles.cache.some((role) => role.name === "Moderator")
+  ) {
+    if (msg.mentions.users.size) {
+      const mentionedUser = msg.mentions.users.first();
+      const user = await userModel.findOrCreateByDiscordId(mentionedUser.id);
+      user.whitelisted ? (user.whitelisted = false) : (user.whitelisted = true);
```
Comment (VuongRon) [inline commands/whitelist.js]: `user.whitelisted = !user.whitelisted`

File: commands/whitelist.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+          .then((response) => {
+            userUsernames.push({
+              username: response.data.username,
+              discriminator: response.data.discriminator,
+            });
+          })
+          .catch((err) => {
+            console.error(err);
+          });
+      }
```
Comment (VuongRon) [inline commands/whitelist.js]: Not a huge deal, but might be cleaner to move the axios stuff into a separate function

File: models/media.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+      },
+      commandName: {
+        type: DataTypes.STRING,
+        allowNull: false,
+      },
+    },
+    {
+      sequelize,
+      modelName: "Media",
+      tableName: "Mediae",
```
Comment (VuongRon) [inline models/media.js:43]: This doesn't seem to match the name of the table?

**Review by VuongRon — COMMENTED** [review]
Looks pretty much good, just a few small considerations.

File: models/media.js (hunk 3 above)
Comment (dcepil) [reply models/media.js:43]: Forgot to change the names in the migration - fixed.

**Review by VuongRon — APPROVED** [review]
peepoFat

**Pull request merged**